internal/juju/client.go

```diff
@@ -248,6 +249,13 @@ func (sc *sharedClient) GetConnection(modelIdentifier *string) (api.Connection,
 // search the modelUUIDCache for the uuid. If it's not found, fill the model
 // cache and try again.
 func (sc *sharedClient) ModelUUID(modelIdentifier string) (string, error) {
+	sc.modelCacheOnce.Do(func() {
```

Comment:
not wile we're doing this migration, but eventually we'd want to verify `modelIdentifier` is a valid UUID

Reply:
here and in related functions